themes/default/content/blog/top-5-things-for-azure-devs-serverless/index.md

```diff
@@ -0,0 +1,211 @@
+---
+title: "Top 5 an Azure Developer Needs to Know: Serverless"
+date: 2021-07-30
+meta_desc: "Building and deploying serverless functions for Azure developers"
+meta_image: azure-top-5.png
+authors:
+    - sophia-parafina
+tags:
+    - azure functions
+    - serverless
+
+---
+
+In the [previous article]({{< relref "/blog/top-5-things-for-azure-devs-vms" >}}), we did a deep dive into virtual machines, first looking at creating and deploying a virtual machine using the Azure Portal, then repeating the process using infrastructure as code. We further demonstrated how to automate provisioning as part of cloud engineering's  build and deploy process.
+
+This article will explore the other end of the cloud infrastructure with serverless, which is an on-demand fully-managed cloud architecture.
+
+<!--more-->
+
+## What is serverless, and when to use it?
+
+Serverless is an event-driven architecture that executes code when triggered. Azure implements serverless with Azure Functions and Events. A function typically performs a single task, such as saving a log file. A function can perform multiple tasks, but in general, functions adhere to the principle of "you only had one job."  Functions can be written in any Azure supported language: C #, F#, JavaScript, PowerShell, Python, and Typescript.
+
+An Event calls a function to execute. It can be a webhook, an HTTP request, or a scheduled trigger. Azure provides many function templates with a range of triggers; furthermore, they can be customized to application requirements.
+
+Azure Functions are ideal for specific use cases that include:
+
+- when demand varies unpredictably, serverless can scale automatically
+- when a service is used infrequently, serverless can reduce costs by only charging per use
+- serverless can eliminate maintenance costs
+- when you have client-heavy applications, where most of the business logic is in the client
+- if you have high latency tasks like data processing
+
+Serverless is not ideal if the application runs constantly, needs a quick response (otherwise known as the [cold start problem]({{< relref "/blog/aws-lambda-provisioned-concurrency-no-cold-starts">}})), or is a long-running process that can exceed the [timeout period](https://docs.microsoft.com/en-us/azure/azure-functions/functions-scale#timeout) for your execution plan.
+
+With those caveats, let's jump into creating a serverless function using the Azure Portal to understand how to write a function and how the pieces work together.
+
+## Our first serverless function
+
+For this example, we'll use [VS Code](https://code.visualstudio.com/) and the [Azure Functions extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azurefunctions).
```

Comment:
```suggestion
For this example, we'll use [Visual Studio Code](https://code.visualstudio.com/) and the [Azure Functions extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azurefunctions).
```

I think that it's preferred to refer to it this way; alternately it's probably "VSCode"?